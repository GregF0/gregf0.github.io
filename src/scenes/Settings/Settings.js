import React from 'react'
import styled, { css } from 'styled-components'
import useMediaQuery from '@material-ui/core/useMediaQuery'
import { Sliders } from './components/Sliders'
import { Switches } from './components/Switches'
import { DaySelect } from './components/DaySelect'

export const Settings = () => {
  const sidenav = +useMediaQuery('(min-width:600px) and (min-height:500px)')

  return (
    <Container sidenav={sidenav}>
      <Sliders />
      <DaySelect />
      <Switches />
    </Container>
  )
}

const Container = styled.div`
  width: 300px;
  margin: auto;
  padding-top: 24px;
  padding-left: 29px;
  padding-right: 32px;

  ${({ sidenav }) =>
    sidenav &&
    css`
      width: 400px;
    `}
`
