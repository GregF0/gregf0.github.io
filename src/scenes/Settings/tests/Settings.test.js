import React from 'react'
import { createShallow } from '@material-ui/core/test-utils'
import { Settings } from '../Settings'

describe('<Settings />', () => {
  const shallow = createShallow()
  const wrapper = shallow(<Settings />)

  test('should render <Settings /> correctly', () => {
    expect(wrapper).toMatchSnapshot()
  })
})
