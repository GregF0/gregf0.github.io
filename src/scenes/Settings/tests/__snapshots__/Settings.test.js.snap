// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Settings /> should render <Settings /> correctly 1`] = `
<styled.div
  sidenav={0}
>
  <Sliders />
  <DaySelect />
  <Switches />
</styled.div>
`;
